assert/assertion_compare.go

```diff
@@ -299,6 +302,27 @@ func compare(obj1, obj2 interface{}, kind reflect.Kind) (CompareType, bool) {
 				return compareLess, true
 			}
 		}
+	// Check for known struct types we can check for compare results.
+	case reflect.Struct:
+		{
+			// All structs enter here. We're not interested in most types.
+			if !canConvert(obj1Value, timeType) {
+				break
+			}
+
+			// time.Time can compared!
+			timeObj1, ok := obj1.(time.Time)
+			if !ok {
+				timeObj1 = obj1Value.Convert(timeType).Interface().(time.Time)
+			}
```

Comment:
Why are we converting to `timeType` (i.e. `time`), then interface, then back to time?

Reply:
`Convert()` returns a `reflect.Value`, `Interface()` takes us _out_ of the reflection wrappers, and then finally we do a type assertion. It looks a bit convoluted, but it's all necessary!